clientview.ts

```
@@ -0,0 +1,41 @@
+import {transact, getMutationID} from './db';
+import {gcal, getEvents} from './gcal';
+
+export async function clientView(clientID: string, auth: string) {
+    const calendars = (await gcal('/users/me/calendarList', auth)).items;
+    console.log(`got ${calendars.length} calendars`);
+
+    let lastMutationID = 0;
+    await transact(async (db) => {
+        lastMutationID = await getMutationID(db, clientID);
+    });
+
+    const events = [];
+    (await Promise.all(calendars.map(async (c: any) => {
+        console.log('Getting events for', c.id);
+        return await getEvents(c.id, auth);
+    }))).forEach((r: Array<any>) => events.push(...r));
+    console.log(`got ${events.length} events`);
+
+    const eventDate = (d: any) => {
```

[cbb] Too many `any` to my liking. `any` hides errors.